### PR TITLE
[mono] Propagate -Werror properly, fix warning

### PR DESCRIPTION
**Pull request by CoffeeFlux**
Apparently `--enable-werror` broke some time back, so fix it.

Contributes to https://github.com/dotnet/runtime/issues/41357

CMake probably needs similar changes. cc: @vargaz

### CODE REVIEW

**Comment by CoffeeFlux** [comment]
It's possible I will also need to disable `-Werror` for LLVM; we'll have to see if the lanes pass.

**Comment by vargaz** [comment]
Shouldn't -Werror be added to the normal CFLAGS ?

**Comment by naricc** [comment]
If this is prone to breaking, is there any way to write a test for it? Like, a test we expect not to build or something?

**Comment by CoffeeFlux** [comment]
@vargaz the commit making it separate indicated that putting it in CFLAGS early in the build script broke some things as part of the config. I can try but I'm not optimistic.

**Comment by vargaz** [comment]
Try adding it to SHARED_CFLAGS, thats not used by configure checks.

File: src/mono/mono/metadata/cominterop.c [hunk 1]
```diff
@@ -4155,6 +4155,7 @@ ves_icall_System_Runtime_InteropServices_Marshal_QueryInterfaceInternal (MonoIUn
 #endif /* HOST_WIN32 */
 #endif /* DISABLE_COM */
 
+#ifndef ENABLE_NETCORE
 MonoStringHandle
```
Comment (vargaz) [inline src/mono/mono/metadata/cominterop.c:4159]: Is this related ?

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/metadata/cominterop.c (hunk 1 above)
Comment (CoffeeFlux) [reply src/mono/mono/metadata/cominterop.c:4159]: Yes, it's a warning that slipped in because this wasn't actually working.
